app/src/main/java/com/jjswigut/matters/ui/MatterListRecyclerViewAdapter.kt

```diff
@@ -23,104 +22,32 @@ class MatterListRecyclerViewAdapter :
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val item = values[position]
+        val item = matters[position]
         holder.titleView.text = item.matterTitle
         holder.contentView.text = item.matterContent
+
+//        holder.itemView.setOnClickListener {
```

Comment:
so the problem of passing along clicks from a recyclerview to a fragment is pretty common! one way you can make it work is pass in a lambda in your adapter's constructor

so picture something like this

```kotlin
sealed class MatterAction {
    data class MatterClicked(val position: Int, val matter: Matter) : MatterAction()
}
```

and if you wanna get really fancy you can use something called a typealias to make a type out of our lambda, so something like

```typealias MatterActionHandler = (MatterAction) -> Unit```

so we are essentially creating our own type (but just an alias for it) that says if you encounter a lambda that takes in a `MatterAction` and returns nothing, then you can just use this new type instead.

so now your adapter can look something like

class `Adapter(private val stuff: List<Stuff>, private val actionHandler : MatterActionHandler)`

and then when you create your Adapter you can do something like

```kotlin
fun handleAction(action : MatterAction) {

}
```

```kotlin
adapter = Adapter(listOf(), ::handleAction)
```

and then to use that action handler in your adapter you can just do 
```kotlin
item.setOnClickListener {
    actionHandler(MatterAction.MatterClicked(position, matter))
}
```

Reply:
I'm like half following.
The sealed class creates a schema for the data of whatever note is clicked. Then does it extend MatterAction? I get confused by all the ":" sometimes.
I want to be fancy but I'm lost about why the typealias is necessary
Then we put the actionHandler in the adapter constructor.
Does handleAction() hold the navigation stuff?
Where should I actually construct the adapter then? This also references the above comment about building it with every view instead of the the fragment. 
```
private var listAdapter:MatterListRecyclerViewAdapter =
            MatterListRecyclerViewAdapter(listOf(), actionHandler)
```
this is what I'm picturing at the top but it obviously doesn't work.

Reply:
yup! so when you create a sealed class, all of the classes inside that sealed class extend the parent class. so in an example like this

```kotlin
sealed class MatterAction {
    data class MatterClicked(val position: Int, val matter: Matter) : MatterAction()
    data class MatterLongClicked(val position: Int) : MatterAction()
}
```

you can use `is a` language on them. so MatterLongClicked is a Matter action. MatterClicked is a MatterAction. You often use `is a` language when talking about object oriented programming languages and inheritance.

the type alias isn't necessary! it's just a simpler way of referring to the lambda that we've created. an alias is just a reference, so it's not necessary. anywhere you could put the alias, you could also put it's real type of (MatterAction) -> Unit

so you'll want to have something like this at the end


```kotlin
class Fragment: Fragment() {
    private lateinit var adapter : Adapter
    
    override fun onCreate() {
        adapter = Adapter(listOf(), ::handleAction)
    }

    fun handleAction(matterAction : MatterAction) {
        when(matterAction) {
            is MatterActionClicked -> {
                // do what you want to do when matter is clicked
            }
        }
    }
} 
````

Reply:
how does listOf() work? When I initialize the adapter in oncreate, I don't really have a list to pass to it because I get the list when I initialize the recyclerview in onViewCreated using the coroutine.